src/Client.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Tgallice\FBMessenger;
+
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\RequestOptions;
+use Psr\Http\Message\ResponseInterface;
+use Tgallice\FBMessenger\Exception\ApiException;
+
+class Client
+{
+    /**
+     * API base uri
+     */
+    const API_BASE_URI = 'https://graph.facebook.com/';
+
+    /*
+     * API Version
+     */
+    const DEFAULT_API_VERSION = 'v2.6';
+
+    /**
+     * Request default timeout
+     */
+    const DEFAULT_TIMEOUT = 60;
+
+    /**
+     * Request default file upload timeout
+     */
+    const DEFAULT_FILE_UPLOAD_TIMEOUT = 3600;
+
+    /**
+     * @var array
+     */
+    public static $allowedMethod = ['POST', 'GET', 'PUT', 'DELETE'];
+
+    /**
+     * @var string Wit app token
+     */
+    private $accessToken;
+
+    /**
+     * @var ClientInterface client
+     */
+    private $client;
+
+    /**
+     * @var ResponseInterface|null
+     */
+    private $lastResponse;
+
+    public function __construct($accessToken, ClientInterface $httpClient = null)
+    {
+        $this->accessToken = $accessToken;
+        $this->client = $httpClient ?: $this->defaultHttpClient();
+    }
+
+    /**
+     * @param string $uri
+     * @param string|null $body
+     *
+     * @return ResponseInterface
+     */
+    public function post($uri, $body = null)
+    {
+        return $this->send('POST', $uri, $body);
```

Comment:
Good catch ! I had set the content-type to `application/json` as default but I have removed it since there had an issue with file upload. I forgot to test again the thread-settings... I will rework this part to a better way.